### PR TITLE
feat: Enhance debug log for order trash and untrash log

### PR DESCRIPTION
**Pull request by mralaminahamed**
### All Submissions:
* [x] My code follows the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [x] My code satisfies feature requirements
* [x] My code is tested
* [x] My code passes the PHPCS tests
* [x] My code has proper inline documentation
* [ ] I've included related pull request(s) (optional)
* [x] I've included developer documentation
* [x] I've added proper labels to this pull request

### Changes proposed in this Pull Request:
- Added new `log_message` method for standardized logging with error levels
- Created common `process_order_status` method to handle both trash/untrash operations
- Improved error handling and code organization
- Reduced code duplication by consolidating similar functionality

### Related Pull Request(s)
* N/A

### Closes 
* N/A - Enhancement PR

### How to test the changes in this Pull Request:
1. Install and activate latest Dokan Pro
2. Go to WooCommerce > Orders
3. Test trashing an order:
   - Select an order and move it to trash
   - Check logs in `WooCommerce/Status/Logs` > dokan
   - Verify log format includes level and proper message
4. Test restoring an order:
   - Go to trash
   - Restore an order
   - Check logs again
   - Verify database status is updated correctly
5. Verify proper error handling:
   - Try with invalid order ID (can be simulated in code)
   - Check error logs format

Pattern:
 1. user: 
  [Order Sync] Order %d has been %s moved to trash by  %s (%s) from IP: %s.
  [Order Sync] Update to order #%d (Status: %s) in Dokan tables. Orders: %s (OK|FAILED), Vendor Balance: %s (OK|FAILED)
 2. system:
  [Order Sync] Order %d has been %s moved to trash by System.
  [Order Sync] Update to order #%d (Status: %s) in Dokan tables. Orders: %s (OK|FAILED), Vendor Balance: %s (OK|FAILED)

Examples:
1. user: 
 [Order Sync] Order #2874 moved to trash by We ba Shope2e (administrator) from IP: 127.0.0.1
 [Order Sync] Update order #2822 (Status: wc-trash) in Dokan tables. Orders: OK, Vendor Balance: OK
2. system:
 [Order Sync] Order 3478 has been %s moved to trash by System.
 [Order Sync] Update to order #3478 (Status: wc-trash) in Dokan tables. Orders: OK, Vendor Balance: OK
 

CC: @StalinDurjo vai, @shashwatahalder01 vai

### Changelog entry
*feat: Enhance debug log for order trash and untrash operations*

Improved order event logging with standardized format and better error handling. Consolidated trash/untrash operations for better code maintainability. Added user context information in logs for better traceability.

### Before Changes
Order trash and untrash operations had duplicate code and inconsistent logging:

```php
public function after_order_trash( int $order_id ) {
    // Duplicate error handling
    // Basic logging without levels
    dokan_log( "Failed to fetch order {$order_id} during trash operation." );
}

public function after_order_untrash( int $order_id ) {
    // Same duplicate code
    dokan_log( "Failed to fetch order {$order_id} after untrash." );
}
```

### After Changes
Consolidated code with improved logging:

```php
public function process_order_status( int $order_id ): void {
    $action = current_action() === 'woocommerce_trash_order' ? 'trashed' : 'restored';
    ...... 

   $this->process_order_status( $order, $wpdb, $order_id );
    $this->log_message( '...', $order_id );
}

private function log_message( string $action, int $order_id ): void {
    .....
}
```

### Feature Video
N/A - Code enhancement PR

### PR Self Review Checklist:
* [x] Code follows style guidelines - Verified against WPCS
* [x] Clear naming - Methods and variables are self-descriptive
* [x] KISS - Simplified code by removing duplication
* [x] DRY - Consolidated duplicate code into single method
* [x] Code is readable - Improved structure and error handling
* [x] No performance issues - Same operations as before
* [x] Self-explanatory code with proper comments
* [x] No grammar errors in comments and logs

* [ ] Correct — Does the change do what it's supposed to?
* [ ] Secure — Would a nefarious party find some way to exploit this change?
* [ ] Readable — Will your future self be able to understand this change months down the road?
* [ ] Elegant — Does the change fit aesthetically within the overall style and architecture?

## Summary by CodeRabbit

- **New Features**
	- Enhanced logging for order actions, capturing user details for better traceability.
	- Improved modular structure for order status updates.

- **Documentation**
	- Expanded method documentation for clarity on purposes and parameters.

### CODE REVIEW

File: includes/Order/OrderEventListener.php [hunk 1]
```diff
@@ -50,7 +51,30 @@ public function after_order_trash( int $order_id ) {
...
+			);
+		} else {
+			dokan_log(
+                sprintf(
+                    'Order %d has been trashed in Dokan order and vendor balance tables by System (system).',
+                    $order_id
+                )
+			);
+		}
     }
```
[the remark this answers is not in the document]
Comment (mralaminahamed) [reply includes/Order/OrderEventListener.php]: vai, new logger method created and a new method for common on trash and untrash method

**Comment by mrabbani** [comment]
@mralaminahamed vai, Pls write the testing steps.

**Comment by mralaminahamed** [comment]
Refactored the OrderEventListener class to improve code organization and reduce duplication. The changes focus on better error handling, standardized logging, and consolidated event handling logic.

## What has changed?
- Added new `log_message` method for standardized logging
- Created common `process_order_status` method to handle both trash/untrash operations
- Improved error handling and type safety
- Reduced code duplication
- Enhanced code maintainability

## Why is this change needed?
- Previous implementation had duplicate code in trash and untrash methods
- Logging was inconsistent across different methods
- Error handling needed improvement
- Code organization could be more efficient

## How was this tested?
1.  **Manual Testing:**
   ```
   Test Environment:
   - WordPress 6.4
   - WooCommerce 8.2
   - PHP 8.1
   ```
   
   **Test Cases:**
   - Trashing orders:
     - [x] Single order trash
     - [x] Bulk order trash
     - [x] Orders with different statuses
   
   - Restoring orders:
     - [x] Single order restore
     - [x] Bulk order restore
     - [x] Orders with different statuses
   
   - Logging:
     - [x] Correct log format
     - [x] Proper error messages
     - [x] User action tracking


## Breaking Changes
- None. This is a internal refactoring that maintains all existing functionality

## Dependencies
- No new dependencies added
- Requires PHP 7.4+ (unchanged)
- Compatible with existing WordPress and WooCommerce versions


CC: @shashwatahalder01 vai, @StalinDurjo vai

**Comment by StalinDurjo** [comment]
Issue: User's name is not visible in order trashed and restored logs
Description: If user's name is not set, then in the logs for order trashed and restored, empty spaces are shown.

Attachment
Screenshot 2024-11-01 at 2 30 55 PM

@mralaminahamed bhai

File: includes/Order/OrderEventListener.php [hunk 2]
```diff
@@ -67,31 +54,115 @@ public function after_order_untrash( int $order_id ) {
...
-				'trn_type' => 'dokan_orders',
+                'trn_id'   => $order_id,
+                'trn_type' => 'dokan_orders',
             ),
             array( '%s' ),
             array( '%d', '%s' )
         );
 
-        dokan_log( "Order {$order_id} has been restored in Dokan order and vendor balance tables with status: {$previous_status}" );
+        $this->log_db_update( $order_id, $previous_status, $orders_updated, $balance_updated );
```
Comment (mrabbani) [inline includes/Order/OrderEventListener.php:102]: This log is no longer needed.

**Review by mrabbani — CHANGES_REQUESTED** [review]

File: includes/Order/OrderEventListener.php (hunk 2 above)
Comment (mralaminahamed) [reply includes/Order/OrderEventListener.php]: vai, removed

**Pull request merged**